src/java/org/apache/cassandra/index/sai/SSTableContext.java

```diff
@@ -141,6 +178,20 @@ public PrimaryKeyMap.Factory primaryKeyMapFactory()
         return primaryKeyMapFactory;
     }
 
+    public PrimaryKey minSSTableKey()
+    {
+        if (minSSTableKey == null)
+            throw new IllegalStateException("minSSTableKey is null");
```

Comment:
Suggestion: add a hint they may be null because loading them was skipped (however I'd really prefer we just lazy load them instead of throwing).

Reply:
But then we'd throw them from `DummyThrowingFactory` and it would be a bit more obvious from the stacktrace - by searching `DummyThrowingFactory` you'd immediately find this piece of code above; instead of having to follow the data backwards from the constructor param to understand why loading was blocked.